### DISCUSSION TITLE
Sveltekit + Pocketbase + Marmot = ❤️

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [maxpert/marmot#89] [Show and tell]
Hey everyone,

I've successfully scaled my SvelteKit app horizontally using Pocketbase and Marmot, and it's been smooth sailing.

My micro-SaaS application is a stock analysis platform where users can sign up, create their watchlist, and leverage AI signals for bullish or bearish stocks, all supported by thorough backtesting results.

Additionally, I've developed a community hub similar to Reddit/9GAG where users can share memes, comment, and like posts. This entire venture has been a solo project, and I must say, this setup has worked wonders for scaling my Svelte project.

If you're curious or have any questions, feel free to reach out! 😊

![Screenshot from 2023-11-28 00-11-21](https://github.com/maxpert/marmot/assets/22890183/4536a6ba-be9f-4def-a7b0-fa5570d1bac1)

Thanks for @maxpert for the really kind support to make this happen

Website: https://stocknear.com/

**Comment by burggraf** [comment]
Would you care to share your scripts for scaling Pocketbase with Marmot?

**Reply by MuslemRahimi** [reply]
There are no real scripts in my case. However i can write down the steps to scale Pocketbase with Marmot when you have two different VPS Servers.

1) Create for both a config.toml
```
seq_map_path="/tmp/marmot-1-sm.cbor"
db_path="pb_data/data.db"
node_id=1
```
For your second server it should be node_id=2 and marmot-2-sm.cbor
2) Bypass the firewall of your VPS server
```
sudo ufw allow 4222/tc
```
3) For the last step you just have to connect them but make sure both databases are identical before connecting them
For your first VPS server type  
```
./marmot -config config.toml -cluster-addr <IP-ADDRESS-VPS-1>:4222 -cluster-peers 'nats:/<IP-ADDRESS-VPS-2>:4222/' 
```
and for your second VPS server type  
```
./marmot -config config.toml -cluster-addr <IP-ADDRESS-VPS-2>:4222 -cluster-peers 'nats:/<IP-ADDRESS-VPS-1>:4222/' 
```
4) Voila! It should now work and everything should be in synced

**Reply by MuslemRahimi** [reply]
If any errors or problems are happening I usually follow these commands to clean up
```
rm -rf /tmp/nats && rm -rf /tmp/*.cbor
```
```
./marmot -config config.toml -cleanup
```
I hope these little steps will help you by your project

**Reply by burggraf** [reply]
Awesome, thanks for this!  How do you handle password protection/security between the nodes?  Care to share any other relevant `config.toml` settings that are important here?

**Reply by burggraf** [reply]
Do you create a snapshot first?  If so, can you show your configuration for that?

**Reply by MuslemRahimi** [reply]
Sorry for the late reply.
If you want to send data between two VPS server you have to follow these steps too so that no password is needed anymore

1. ssh-keygen -t rsa
2. ssh-copy-id root@<IP-ADDRESS-VPS-2>
3. chmod 600 /root/.ssh/id_rsa
4. Test if data can be send: ping <IP-ADDRESS-VPS-2>


Do the same exact steps for your other server but this time replace every IP-Address with <IP-ADDRESS-VPS-1>

**Reply by MuslemRahimi** [reply]
I don't create snapshots. Sorry I can't help you with that

**Reply by MuslemRahimi** [reply]
The only information I put on my config.toml file was
```
seq_map_path="/tmp/marmot-1-sm.cbor"
db_path="pb_data/data.db"
node_id=1
```
but i am sure @maxpert can give you more information if you need some help.
For faster response time definitely try to reach out to him via [Discord](https://discord.gg/STrCPmX2yj)
